src/lib.rs

```diff
@@ -6,13 +6,165 @@ pub mod scheduled_tasks;
 #[cfg(feature = "console")]
 pub mod telemetry;
 
-use lemmy_utils::error::LemmyError;
+use crate::{code_migrations::run_advanced_migrations, root_span_builder::QuieterRootSpanBuilder};
+use actix_web::{middleware, web::Data, App, HttpServer, Result};
+use doku::json::{AutoComments, CommentsStyle, Formatting, ObjectsStyle};
+use lemmy_api_common::{
+  context::LemmyContext,
+  lemmy_db_views::structs::SiteView,
+  request::build_user_agent,
+  utils::{
+    check_private_instance_and_federation_enabled,
+    local_site_rate_limit_to_rate_limit_config,
+  },
+  websocket::chat_server::ChatServer,
+};
+use lemmy_db_schema::{
+  source::secret::Secret,
+  utils::{build_db_pool, get_database_url, run_migrations},
+};
+use lemmy_routes::{feeds, images, nodeinfo, webfinger};
+use lemmy_utils::{
+  error::LemmyError,
+  rate_limit::RateLimitCell,
+  settings::{structs::Settings, SETTINGS},
+};
+use reqwest::Client;
+use reqwest_middleware::ClientBuilder;
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+use reqwest_tracing::TracingMiddleware;
+use std::{env, sync::Arc, thread, time::Duration};
 use tracing::subscriber::set_global_default;
+use tracing_actix_web::TracingLogger;
 use tracing_error::ErrorLayer;
 use tracing_log::LogTracer;
 use tracing_subscriber::{filter::Targets, layer::SubscriberExt, Layer, Registry};
 use url::Url;
 
+/// Max timeout for http requests
+const REQWEST_TIMEOUT: Duration = Duration::from_secs(10);
+
+/// Placing the main function in lib.rs allows other crates to import it and embed Lemmy
+pub async fn main2() -> Result<(), LemmyError> {
```

Comment:
Maybe rename to something other than main2

Reply:
done